include/sundials/sundials_memory.h

```diff
@@ -65,6 +66,10 @@ struct _SUNMemoryHelper
 {
   void*               content;
   SUNMemoryHelper_Ops ops;
+  unsigned long long  num_allocations;
+  unsigned long long  num_deallocations;
```

Comment:
Objects currently clone the helper when they are cloned, so won't the helper stats be spread across the different helper clones in the cloned objects?

Reply:
Yes, it is specific to each helper and I don’t see a way around that. Perhaps objects shouldn’t clone the helper when they are cloned? But then if some implementation needed it to be cloned that would be a problem. The amrex helper implementation of clone just returns the singleton, so it’s always the same instance.

Reply:
Could we get around this by clarifying the issue in our documentation, and recommending that users who supply a custom helper should follow the "singleton" approach?  We could add a warning that for custom helpers that *do not follow* the recommendation, then these inquiry routines will be inaccurate.

Reply:
I wouldn’t say that the routines are inaccurate, just that the stats are specific to the instance of the helper and someone may not expect that we create clones under the hood.  We can add a warning that emphasizes this for sure.

Reply:
Also we can suggest that this behavior can be modified by overriding the clone operation, and then show a singleton pattern as an example.